### PR TITLE
Implement QuoteIdentifier and UnquoteIdentifier in OdbcCommandBuilder

### PR DESCRIPTION
**Pull request by EgorBo**
`QuoteIdentifier(..)` and `UnquoteIdentifier(..)` in `OdbcCommandBuilder` are not implemented and throw NIE - I copy-pasted code from [reference-sources](http://referencesource.microsoft.com/#System.Data/System/Data/Odbc/OdbcCommandBuilder.cs,240) to fill the gaps and added tests. It also fixes few tests in mono as part of https://github.com/mono/mono/pull/4893

### CODE REVIEW

File: src/System.Data.Odbc/src/System/Data/Odbc/OdbcCommandBuilder.cs [hunk 1]
```diff
@@ -273,9 +306,46 @@ public override string UnquoteIdentifier(string quotedIdentifier)
...
+
+            // if the user has specificed a prefix use the user specified  prefix and suffix
+            // otherwise get them from the provider
+            string quotePrefix = QuotePrefix;
+            string quoteSuffix = QuoteSuffix;
+            if (string.IsNullOrEmpty(quotePrefix))
+            {
+                if (connection == null)
+                {
+                    // VSTFDEVDIV 479567: use the adapter's connection if UnquoteIdentifier was called from
```
Comment (danmosemsft) [inline src/System.Data.Odbc/src/System/Data/Odbc/OdbcCommandBuilder.cs]: I would remove legacy bug numbers...

File: src/System.Data.Odbc/src/Resources/Strings.resx [hunk 2]
```diff
@@ -201,6 +201,9 @@
   <data name="ADP_NonPooledOpenTimeout" xml:space="preserve">
     <value>Timeout attempting to open the connection.  The time period elapsed prior to attempting to open the connection has been exceeded.  This may have occurred because of too many simultaneous non-pooled connection attempts.</value>
   </data>
+  <data name="ADP_QuotePrefixNotSet" xml:space="preserve">
+    <value>{0} requires open connection when the quote prefix has not been set.</value>
```
Comment (danmosemsft) [inline src/System.Data.Odbc/src/Resources/Strings.resx]: nit: "an open connection" ?

**Comment by danmosemsft** [comment]
It looks like our Unix machines are not provisioned with the ODBC dependency yet: https://github.com/dotnet/corefx/issues/15776
Also, the tests should be disabled on Windows Nano as it does not have odbc.dll.

I think it's OK to continue to make changes like this one based on running the tests locally, but CI needs to pass ie you need to disable the tests on platforms they aren't going to pass on right now.

**Comment by karelz** [comment]
@EgorBo are you still working on the PR? What is the next step?

**Comment by EgorBo** [comment]
@karelz you mean this PR? it's finished I only had to disable my tests (like the other odbc tests are) as @danmosemsft suggested.

**Comment by karelz** [comment]
OK, then we are blocked on code review I guess - @corivera @divega @saurabh500 can you please help?

**Comment by danmosemsft** [comment]
ping @corivera @divega @saurabh500

File: src/System.Data.Odbc/src/System/Data/Odbc/OdbcCommandBuilder.cs [hunk 3]
```diff
@@ -244,16 +244,49 @@ private static OdbcParameter[] DeriveParametersFromStoredProcedure(OdbcConnectio
...
+            // otherwise get them from the provider
+            string quotePrefix = QuotePrefix;
+            string quoteSuffix = QuoteSuffix;
+            if (string.IsNullOrEmpty(quotePrefix))
+            {
+                if (connection == null)
+                {
+                    // Use the adapter's connection if QuoteIdentifier was called from 
+                    // DbCommandBuilder instance (which does not have an overload that gets connection object)
+                    connection = DataAdapter?.SelectCommand?.Connection;
```
Comment (corivera) [inline src/System.Data.Odbc/src/System/Data/Odbc/OdbcCommandBuilder.cs]: Looks like this line in the reference source is "connection = base.GetConnection() as OdbcConnection; "

Do we not have GetConnection() in Core?

Comment (corivera) [inline src/System.Data.Odbc/src/System/Data/Odbc/OdbcCommandBuilder.cs]: Looked at the reference source, and the connection retrieval logic is the same. Ignore this comment.

**Comment by karelz** [comment]
Merging as we have tests passing and area expert code review.

**Pull request merged**